### PR TITLE
feature: support specifying env-vars when creating model from model package

### PR DESCRIPTION
**Pull request by saumitravikram**
*Issue #, if available:* https://github.com/aws/sagemaker-python-sdk/pull/3301

*Description of changes:* Changes made to enable back specifying of environment variable when creating model from model package. Previously service use to throw exception *Environment variable map cannot be specified both at Model Package and Model* thus above change was made in python sdk to exclude specifying environment variable. Now service is supporting the same and we are including back the change.

*Testing done:*
Unit testing
tox -e py38 tests/unit/

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I certify that the changes I am introducing will be backward compatible, and I have discussed concerns about this, if any, with the Python SDK team
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [ ] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have added unit and/or integration tests as appropriate to ensure backward compatibility of the changes
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [x] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/unit/sagemaker/model/test_model_package.py [hunk 1]
```diff
@@ -52,6 +52,10 @@
     "CertifyForMarketplace": False,
 }
 
+ENV_KEY_1 = "env_key_1"
+ENV_VALUE_1 = "env_key_1"
+ENVIRONMENT = {ENV_KEY_1: ENV_VALUE_1}
+
```
Comment (rahven14) [inline tests/unit/sagemaker/model/test_model_package.py:58]: If these 3 variables are only used in test_create_sagemaker_model_include_environment_variable function, then the scope can be local to the function itself, hence we can move these constants within the function itself.

File: tests/unit/sagemaker/model/test_model_package.py [hunk 2]
```diff
@@ -52,6 +52,10 @@
     "CertifyForMarketplace": False,
 }
 
+ENV_KEY_1 = "env_key_1"
+ENV_VALUE_1 = "env_key_1"
```
[the remark this answers is not in the document]
Comment (saumitravikram) [reply tests/unit/sagemaker/model/test_model_package.py:56]: updating

File: tests/unit/sagemaker/model/test_model_package.py (hunk 1 above)
Comment (saumitravikram) [reply tests/unit/sagemaker/model/test_model_package.py:58]: updating

**Review by rahven14 — APPROVED** [review]
Approval from SRS team

**Review by knikure — COMMENTED** [review]
/bot run slow-tests

**Review by knikure — APPROVED** [review]

**Pull request merged**